### ISSUE TITLE
URL safety/validation - http://www.dilbert.com/rss/

### ISSUE THREAD

**Issue by RafalBabinicz** [skeeto/elfeed#17]
Scenario:
Boring day 1.
I want to add some daily humor feed and I go to http://www.dilbert.com/
I see a standard RSS img link (http://www.dilbert.com/rss/). 
I copy it and paste directly to my *elfeed-feeds-alist*.

Regular day 2.
I open my beloved elfeed and want to update my feeds, 
but in the middle of the update process my CPU workload goes to 100% and my available RAM is gradually smaller and smaller - after 3GB nothing left.

And I totally forget about new feed addition. Anyway - what can be wrong with 
*http://www.dilbert.com/rss/*?

But it reality - everything is wrong.
http://www.dilbert.com/rss/ is an address where you can find proper RSS links :)
Not a real feed url.

After update I get "odd number" #13 surprise.
Fortunately I have a cronjob for .elfeed dir backup.
Great success! I do not lost my data.

In conclusion:
*Elfeed should be able to detect invalid feeds/formats and for sure do not kill RAM and CPU, b/c of them, during update.*

Now I will always check URL manually, but such safety mechanism would be nice addition.

Tested with:
URL: http://www.dilbert.com/rss/
Emacs 24.3, emacs 24.4
Elfeed from master HEAD.

**Comment by RafalBabinicz** [comment]
Fixed in emacs’ `emacs-25` and `master` branches ヽ(*＾▽＾)／
[8b01e69 | * Prevent infinite loop on not-well-formed xml. (Bug#16344)](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=16344#8)
